tests/ui/sanitizer/cfi-complex-receiver.rs

```diff
@@ -2,11 +2,16 @@
 // * Arc<dyn Foo> as for custom receivers
 // * &dyn Bar<T=Baz> for type constraints
 
-//@ needs-sanitizer-cfi
+//@ revisions: cfi kcfi
 // FIXME(#122848) Remove only-linux once OSX CFI binaries work
 //@ only-linux
-//@ compile-flags: --crate-type=bin -Cprefer-dynamic=off -Clto -Zsanitizer=cfi
-//@ compile-flags: -C target-feature=-crt-static -C codegen-units=1 -C opt-level=0
+//@ [cfi] needs-sanitizer-cfi
+//@ [kcfi] needs-sanitizer-kcfi
+//@ compile-flags: -C opt-level=0
+//@ [cfi] compile-flags: -C target-feature=-crt-static -C codegen-units=1 -C lto
+//@ [cfi] compile-flags: -C prefer-dynamic=off
```

Comment:
Hmm. Why are these on two lines exactly? And why can kcfi skip these?

Reply:
`-C codegen-units=1` and `-C lto` are required for CFI in order to have global knowledge of type membership. Not requiring them is one of the reasons people choose to use KCFI instead. We could theoretically use `-C linker-plugin-lto` instead of `-C lto` instead, which allows more codegen units, but might requires us to set `-Zsplit-lto-unit`. If you don't like `codegen-units=1` I can try to play with that, but some form of LTO will be required regardless.

`-C target-feature=-crt-static` is because musl defaults to `crt-static`, which is incompatible with most Linux sanitizer implementations, and so is explicitly [disallowed](https://github.com/rust-lang/rust/blob/5f2c7d2bfd46cad00352ab7cd66242077e2e518c/compiler/rustc_session/src/session.rs#L1227). We might be able to loosen that, but it should be a different change.

Arguably, `-crt-static` should be set for `kcfi` as well, but I haven't enabled it for the musl target so it's not yet an issue, and it doesn't need a runtime, so if we removed the configuration suppression, it would just work with `+crt-static` on musl.

`-C prefer-dynamic=off` is disabled because [DSO support](https://clang.llvm.org/docs/ControlFlowIntegrity.html#shared-library-support) is special/complex in CFI. In KCFI, since we don't need global knowledge, we don't care about this.

Reply:
I don't mind these settings, I just wanted to establish the purpose of them for reference.

I would prefer setting `-crt-static` for both versions, yes, even if musl doesn't work with kcfi yet, to indicate that it isn't necessarily a thing about either one.

Reply:
Done.